src/MainWindow.vala

```diff
@@ -816,54 +747,61 @@ namespace Terminal {
                 terminal_widget.run_program (program, location);
             }
 
-            save_opened_terminals (true, true);
+            terminal_widget.child_exited.connect (() => {
+                if (!terminal_widget.killed) {
```

Comment:
I think this will cause a memory leak since the lambda keeps a reference on us and the terminal widget so neither of them will be freed :(

See https://gitlab.gnome.org/GNOME/vala/-/issues/1548